packaging/mgmtworker/files/opt/mgmtworker/scripts/fetch-logs

```diff
@@ -0,0 +1,117 @@
+            f'{contents.status_code}- {contents.reason}\n'
+        )
+        return
+
+    files = [entry['name'] for entry in contents.json()
+             if entry['type'] == 'file']
+
+    dirs = [entry['name'] for entry in contents.json()
+            if entry['type'] == 'directory']
+
+    for file_name in sorted(files):
+        downloaded_file = session.get(url + file_name)
+        file_path = path + file_name
+
+        if downloaded_file.status_code == 200:
+            log_handle.write(f'Retrieved file {file_path}\n')
+            with open(os.path.join(save_path, file_name), 'w',
+                      encoding='utf8') as file_handle:
+                file_handle.write(downloaded_file.text)
+        else:
+            log_handle.write(
+                f'Failed to retrieve file {file_path}: '
+                f'{downloaded_file.status_code}- {downloaded_file.reason}\n'
+            )
+
+    for dir_name in sorted(dirs):
+        get_logs_dir(server, path + dir_name + '/',
+                     data_dir, log_handle, session)
+
+
+def get_logs(servers, ca_cert, credentials):
+    try:
+        temp_path = mkdtemp()
+        zip_file = NamedTemporaryFile(prefix='cfylogs', suffix='.zip',
+                                      delete=False)
+        zip_file.close()
+        zip_path = zip_file.name
+
+        username = credentials['username']
+        password = credentials['password']
+
+        for server in servers:
+            session = requests.Session()
+            session.auth = (username, password)
+            session.verify = ca_cert
+
+            with open(os.path.join(temp_path, server + '.log'),
+                      'w') as log_handle:
+                get_logs_dir(server, '', temp_path, log_handle, session)
+
+        make_zip64_archive(zip_path, temp_path)
+        return zip_path
+    finally:
+        shutil.rmtree(temp_path)
+
+
+if __name__ == '__main__':
+    parser = argparse.ArgumentParser(
+        description='Download logs from cloudify nodes',
+    )
+
+    parser.add_argument(
+        '-a', '--addresses',
+        help='Comma separated list of node addresses to collect logs from.',
+        required=True,
+    )
+
+    parser.add_argument(
+        '-c', '--ca-cert',
+        help='Path of the monitoring CA cert.',
+        default='/etc/cloudify/ssl/monitoring_ca_cert.pem',
+    )
+
+    args = parser.parse_args()
+
+    # Credentials provided by env var to avoid password in ps output
+    creds = {'username': '', 'password': ''}
+    for cred in sorted(creds):
+        env_var = f'MONITORING_{cred.upper()}'
+        cred_value = os.environ.get(env_var)
+        if not cred_value:
+            sys.stderr.write(
+                f'Expected {cred} to be provided in env var {env_var}.\n'
+            )
+        creds[cred] = cred_value
+
+    if not all(val for val in creds.values()):
+        sys.stderr.write('Credentials not provided correctly.\n')
+        sys.exit(1)
```

Comment:
ha, you know how much simpler would this be, if you just did `creds = {'username': os.environ['MONITORING_USERNAME']}`

Reply:
Right, but I wanted it to actually log all the missing env vars rather than complaining one thing at a time because that's really annoying.